src/lint_cmd.go

```diff
@@ -102,10 +102,10 @@ func (c *LintCmd) Execute(args []string) error {
 						// Infer source unit name from file path (the
 						// path components after .srclib-cache until
 						// the basename).
-						pcs := strings.Split(absPath, string(os.PathSeparator))
+						pcs := strings.Split(absPath, string(filepath.Separator))
 						for i, pc := range pcs {
 							if pc == buildstore.BuildDataDirName && len(pcs) > i+2 {
-								unitName = filepath.Clean(strings.Join(pcs[i+2:len(pcs)-1], "/"))
+								unitName = filepath.Clean(strings.Join(pcs[i+2:len(pcs)-1], string(filepath.Separator)))
```

Comment:
Perhaps use `filepath.Join(x...)` instead of `strings.Join(x, string(filepath.Separator))` here?